share/poly_test.go

```diff
@@ -317,3 +318,132 @@ func TestPriPolyCoefficients(test *testing.T) {
 	require.Equal(test, a.coeffs, b.coeffs)
 
 }
+
+func TestRefreshDKG(test *testing.T) {
+	g := edwards25519.NewBlakeSHA256Ed25519()
+	n := 10
+	t := n/2 + 1
+
+	// Run an n-fold Pedersen VSS (= DKG)
+	priPolys := make([]*PriPoly, n)
+	priShares := make([][]*PriShare, n)
+	pubPolys := make([]*PubPoly, n)
+	pubShares := make([][]*PubShare, n)
+	for i := 0; i < n; i++ {
+		priPolys[i] = NewPriPoly(g, t, nil, g.RandomStream())
+		priShares[i] = priPolys[i].Shares(n)
+		pubPolys[i] = priPolys[i].Commit(nil)
+		pubShares[i] = pubPolys[i].Shares(n)
+	}
+
+	// Verify VSS shares
+	for i := 0; i < n; i++ {
+		for j := 0; j < n; j++ {
+			sij := priShares[i][j]
+			// s_ij * G
+			sijG := g.Point().Base().Mul(sij.V, nil)
+			// Eval poly
+			prod := pubShares[i][j]
+			require.Equal(test, sijG.String(), prod.V.String())
```

Comment:
This would seem to be a bit closer to what you actually want to test: `require.True(t, sijG.Equal(prod.V))`

And the other places below like this.

Reply:
Yeah true, I'll fix that.